### PR TITLE
Utilize shortmess option to truncate long messages

### PR DESCRIPTION
**Pull request by liuchengxu**

### CODE REVIEW

**Comment by autozimu** [comment]
Maybe I did something stupid.

When I'm trying this out, long messages are not displayed at all.

Can you double check?

screen shot 2018-08-07 at 18 28 03
screen shot 2018-08-07 at 18 28 16

**Comment by liuchengxu** [comment]
Hmm, I can reproduce the diagnostics messages are gone.



but I also believe `s:Truncate()` works as expected.



What's more, GotoDefinition is good:



```vim
call plug#begin('~/.vim/plugged')

Plug 'autozimu/LanguageClient-neovim', {
    \ 'branch': 'next',
    \ 'do': 'bash install.sh',
    \ }

call plug#end()

augroup filetype_rust
    autocmd!
    autocmd BufReadPost *.rs setlocal filetype=rust
augroup END

" Always draw sign column. Prevent buffer moving when adding/deleting sign.
set signcolumn=yes

let g:LanguageClient_serverCommands = {
    \ 'rust': ['rustup', 'run', 'nightly', 'rls'],
    \ }
let $RUST_BACKTRACE = 1
let g:LanguageClient_loggingFile = '/tmp/LanguageClient.log'
let g:LanguageClient_loggingLevel = 'INFO'
let g:LanguageClient_serverStderr = '/tmp/LanguageServer.log'

nnoremap gd :call LanguageClient#textDocument_definition()<CR>

set noswapfile
```

So, the problem is perhaps related to how/when the linter messages will echo? I haven't found the root cause :(.

**Comment by autozimu** [comment]
Yes, it's probably caused by something else. But still we needs to figure out why.

**Comment by autozimu** [comment]
I think the problem is because with `shortmess+=T`, `echo someLongMessage` won't be displayed.

Can you revert the changes made to `s:EchoEllipsis`?

File: autoload/LanguageClient.vim [hunk 1]
```diff
@@ -6,30 +6,36 @@ function! s:Echo(message) abort
...
-        let maxlen -= 18
-    endif
-    if len(a:message) < l:maxlen
-        let l:message = a:message
-    else
-        let l:message = a:message[:l:maxlen - 3] . '...'
-    endif
-    return l:message
+" Credit: ALE, snippets from ale#cursor#TruncatedEcho()
+function! s:Truncate(message, cmd) abort
```
Comment (autozimu) [inline autoload/LanguageClient.vim]: Can you add also the replacing of line endings?

Comment (liuchengxu) [reply autoload/LanguageClient.vim]: I don't understand very well. What does `the replacing of line endings` mean? Anyway, I have pushed a commit to revert the `s:EchoEllipsis()` related changes. Correct me if anything is not suitable.

Comment (autozimu) [reply autoload/LanguageClient.vim]: I meant replacing line endings and tabs from original ALE implementation,

https://github.com/w0rp/ale/blob/9674132933df3e5b126c30da33a26c9be3ef87c6/autoload/ale/cursor.vim#L17

File: autoload/LanguageClient.vim [hunk 2]
```diff
@@ -27,9 +27,31 @@ function! s:EchoEllipsis(message) abort
     echo s:Ellipsis(a:message)
 endfunction
 
+" Credit: ALE, snippets from ale#cursor#TruncatedEcho()
+function! s:Truncate(message) abort
```
Comment (autozimu) [inline autoload/LanguageClient.vim]: Please update this function name or merge the function body into EchomsgEllipsis.

File: autoload/LanguageClient.vim (hunk 1 above)
Comment (liuchengxu) [reply autoload/LanguageClient.vim]: Done.

File: autoload/LanguageClient.vim (hunk 2 above)
Comment (liuchengxu) [reply autoload/LanguageClient.vim]: Done.

**Comment by autozimu** [comment]
Thanks!

**Pull request merged**